Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -288,6 +298,50 @@ void NetworkStorageManager::writeOriginToFileIfNecessary(const WebCore::ClientOr
 #else
     UNUSED_PARAM(didWrite);
 #endif
+
+    if (m_quotaManager)
+        m_quotaManager->originVisited(origin);
+}
+
+NetworkQuotaManager& NetworkStorageManager::quotaManager()
+{
+    if (!m_quotaManager) {
+        Vector<std::pair<WebCore::ClientOrigin, WallTime>> origins;
+        forEachOriginDirectory([&](auto directory) {
+            auto originFile = originFilePath(directory);
+            auto modificationTime = FileSystem::fileModificationTime(originFile);
+            if (modificationTime) {
+                if (auto origin = readOriginFromFile(originFile))
```

Comment:
Should we delete folders for which we are not able to read origin?

Reply:
It's a bit tricky. If an origin file is not written, but the origin folder has data, there's a chance that next time when the origin is visited, the origin file can be created correctly and the data can be used by that origin.

If you think we should remove it, I can file a radar and follow up to remove the folder when origin file does not exist.